docs/src/description/constraints.md

```diff
@@ -34,56 +31,52 @@ public_inputs {
 
 boundary_constraints {
     # these are main constraints.
-    enf a.first = 0
-    enf a.last = 10
-
-    # these are auxiliary constraints, since they are defined against auxiliary trace columns.
-    enf p.first = 1
-    enf p.last = 1
+    enf a.first = 0;
+    enf a.last = 10;
 }
 
 integrity_constraints {
     <omitted for brevity>
 }
 ```
 
-### Public inputs and random values
+### Public inputs
 
-Boundary constraints can access public input values and random values provided by the verifier in their value expressions.
+Boundary constraints can access public input values provided by the verifier in their value expressions.
 
 To use public inputs, the public input must be declared in the `public_inputs` source section. They can be accessed using array indexing syntax, as described by the [accessor syntax rules](./syntax.md#section-specific-accessors).
 
-To use random values, they must be declared in the `random_values` source section. Random values are defined by an identifier which is bound to a fixed-length array (e.g. `rand: [16]`), but it is additionally possible to bind identifiers to specific random values or groups of values (e.g. `rand: [a, b[14], c]`). Random values can be accessed by using `$` followed by the array identifier and the index of the value (e.g. `$rand[10]`) or by using the name of the bound random value or group along with the index of the value within the group (e.g. `a` or `b[5]`), as described by the [accessor syntax rules](./syntax.md#section-specific-accessors).
-
-### Example of boundary constraints with public inputs and random values
+### Example of bus boundary constraints with public inputs
 
-The following is an example of a valid `boundary_constraints` source section that uses public inputs and random values:
+The following is an example of a valid bus `boundary_constraints` source section that uses public inputs:
 
 ```
 def BoundaryConstraintsExample
 
 trace_columns {
-    main: [a, b]
-    aux: [p0, p1]
+    main: [a, b],
 }
 
 public_inputs {
-    stack_inputs: [16]
-    stack_outputs: [16]
+    stack_inputs: [16],
+    stack_outputs: [16],
+}
+
+buses {
+    multiset p,
+    logup q,
 }
 
 boundary_constraints {
     # these are main constraints that use public input values.
-    enf a.first = stack_inputs[0]
-    enf a.last = stack_outputs[0]
-
-    # these are auxiliary constraints that use public input values.
-    enf p0.first = stack_inputs[1]
-    enf p0.last = stack_outputs[1]
+    enf a.first = stack_inputs[0];
+    enf a.last = stack_outputs[0];
 
-    # these are auxiliary constraints that use random values from the verifier.
-    enf b.first = a + $rand[0]
-    enf p1.first = (stack_inputs[2] + $rand[0]) * (stack_inputs[3] + $rand[1])
+    # these are bus constraints that use the `null` identifier to define empty bus constraints
```

Comment:
I would write this as "these are bus constraints that specify that busses must be empty at the beginning and the end of the execution trace"